### PR TITLE
move to verify

### PR DESCRIPTION
**Pull request by SimonCropp**
@clairernovotny this approach does not use pdbs to determine the path. can u verify it work for your deterministic builds?

if it does, and u are happy with this as an approach, i will finish the diffplex logging and u can merge

### CODE REVIEW

**Comment by clairernovotny** [comment]
The approach seems fine to me, thank you! There's a unit test failing though because the `.cs` files are missing a license header.

**Comment by SimonCropp** [comment]
ye fixed the license header

File: Rx.NET/Source/tests/Tests.System.Reactive/Tests/LicenseHeaderTest.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ private int ScanPath(string path, StringBuilder error)
                     || file.Contains("AssemblyInfo.cs")
                     || file.Contains(".Designer.cs")
                     || file.Contains(".Generated.cs")
-                    || file.Contains(".approved.cs")
+                    || file.Contains(".verified.cs")
```
Comment (clairernovotny) [inline Rx.NET/Source/tests/Tests.System.Reactive/Tests/LicenseHeaderTest.cs]: Cool, LGTM!

**Comment by SimonCropp** [comment]
ok. i will add a extension point to Verify so u can intercept the files on failure.

**Comment by SimonCropp** [comment]
unfortunately in the end i had to add the following to the csproj to get this to work

```
<ContinuousIntegrationBuild>false</ContinuousIntegrationBuild>
<Deterministic>false</Deterministic>
```

The reason being is that Deterministic seems to not just clean pdb paths (which breaks approvaltests), it also cleans CallerFilePath attributes (which breaks Verify).

So with that workaround in mind, it is up to you if you want to use Verify or ApprovalTests

**Comment by clairernovotny** [comment]
What's the difference between ApprovalTests or Verify?

**Comment by SimonCropp** [comment]
https://github.com/SimonCropp/Verify/blob/master/docs/compared-to-approvaltests.md 

disclaimer: i am a maintainer on both

**Comment by clairernovotny** [comment]
Cool, I'll stick with Verify. For now, I can set `Deterministic` to false in the Approvals project. Might be useful to think about an alternative way to get the directory needed though, perhaps with build-time env variables. You can detect deterministic paths by looking for `/_/` as the root and take action accordingly.

**Pull request merged**